cmd/package_managers.go

```diff
@@ -129,23 +125,24 @@ type rubyGemsSearchResults struct {
 
 // Gets the GitHub repository URL for the npm package.
 func fetchGitRepositoryFromNPM(packageName string, packageManager pmc.Client) (string, error) {
-	npmSearchURL := "https://registry.npmjs.org/-/v1/search?text=%s&size=1"
+	npmSearchURL := "https://registry.npmjs.org/%s/latest"
+
 	resp, err := packageManager.Get(npmSearchURL, packageName)
 	if err != nil {
 		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to get npm package json: %v", err))
 	}
 
 	defer resp.Body.Close()
-	v := &npmSearchResults{}
+	v := &npmResult{}
 	err = json.NewDecoder(resp.Body).Decode(v)
 	if err != nil {
 		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to parse npm package json: %v", err))
 	}
-	if len(v.Objects) == 0 {
+	if resp.StatusCode == 404 || v.Repository.URL == "" {
 		return "", sce.WithMessage(sce.ErrScorecardInternal,
 			fmt.Sprintf("could not find source repo for npm package: %s", packageName))
 	}
```

Comment:
since this deals with the status code, should we check this before decoding?

Reply:
although I see now the second part needs the decoded fields. so feel free to ignore this.

If you happen to be fixing the other comment, can you use `http.StatusNotFound` instead of 404?

Reply:
Sure